### ISSUE TITLE
Starting with timostamm-protobuf-ts@v2.9.4 the "output_javascript" option no longer emits .d.ts files

### ISSUE THREAD

**Issue by sourishkrout** [bufbuild/plugins#1162]
Superseding [https://github.com/timostamm/protobuf-ts/issues/643#issuecomment-2043191475](https://github.com/timostamm/protobuf-ts/issues/643#issuecomment-2043191475) since the issue does not occur in the upstream project.

Starting with [https://github.com/bufbuild/plugins/tree/main/plugins/community/timostamm-protobuf-ts/v2.9.4](https://github.com/bufbuild/plugins/tree/main/plugins/community/timostamm-protobuf-ts/v2.9.4) when I generate code for the protos here: [https://github.com/stateful/runme/tree/main/internal/api](https://github.com/stateful/runme/tree/main/internal/api) important `*.d.ts` files are being omitted. A downgrade to `v2.9.3` solves the issue. Please note, if I use the upstream project directly (outside of Buf), I can not reproduce the issue.

Happy to refactor my proto files if that will fix the issue but I don't have an inclination how to do that. Please advise! Thank you.

Here's one of the key differences, `v2.9.4`; please note how the `parser_pb.d.ts` is absent:

```
internal/gen/proto/ts/runme/parser
└── v1
    ├── parser_pb.client.d.ts
    ├── parser_pb.client.js
    └── parser_pb.js

2 directories, 3 files
```

And here's `v2.9.3`:

```
internal/gen/proto/ts/runme/parser
└── v1
    ├── parser_pb.client.d.ts
    ├── parser_pb.client.js
    ├── parser_pb.d.ts
    └── parser_pb.js

2 directories, 4 files
```

**Comment by sourishkrout** [comment]
@pkwarren yes, the public BSR would be https://buf.build/stateful/runme/activity/

Namely: "npm install @buf/stateful_runme.community_timostamm-protobuf-ts@latest" will pull in a `v2.9.4` bundled version without `*.d.ts` whereas "npm install @buf/stateful_runme.community_timostamm-protobuf-ts@2.9.3-20240406201323-8223854c0c39.1` continues to work and includes those files.

I can't seem to reopen this issue.

**Comment by pkwarren** [comment]
@sourishkrout - This should be resolved with version v2.9.4, revision 3 of the plugin. I've verified with code generation locally that these previously missing files are now found:
```
google/protobuf/descriptor.d.ts
google/protobuf/wrappers.d.ts
runme/parser/v1/parser.d.ts
runme/runner/v1/runner.d.ts
runme/runner/v2alpha1/runner.d.ts
```

**Comment by sourishkrout** [comment]
Thank you! Much appreciated.
